coding-standard/action.yml

```diff
@@ -0,0 +1,120 @@
+name: "Coding Standard"
+author: "Graycore"
+description: "A Github Action that runs the Magento Coding Standard."
+
+inputs:
+  php_version:
+    required: true
+    default: "8.1"
+    description: "PHP version used to do the coding standard check."
+
+  composer_version:
+    required: true
+    default: "2"
+    description: "The version of composer to use."
+
+  path:
+    required: true
+    default: "app/code"
+    description: "The directory (relative to the project root) in which the coding standard will be checked. Used when the event is not a pull request."
+
+  version:
+    required: false
+    default: "31"
+    description: "The version of the coding standard to use. If not provided, will use the latest version."
+
+  severity:
+    required: false
+    default: "5"
+    description: "The minimum severity required to display an error or warning (default: 5)"
+
+  warning_severity:
+    required: false
+    default: "5"
+    description: "The minimum severity required to display a warning"
+
+  error_severity:
+    required: false
+    default: "5"
+    description: "The minimum severity required to display an error"
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout Project
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        path: standard
+
+    - name: Set PHP Version
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ inputs.php_version }}
+        tools: composer:v${{ inputs.composer_version }}
+        coverage: none
+
+    - name: Install Coding Standard && Codesniffer baseline
+      shell: bash
+      working-directory: standard
+      run: |
+        git config --global advice.detachedHead false
+        composer require "magento/magento-coding-standard"
+        composer config --no-plugins allow-plugins.digitalrevolution/php-codesniffer-baseline true
+        composer require --dev "digitalrevolution/php-codesniffer-baseline"
+
+
+    - name: Register Coding Standard
+      shell: bash
+      working-directory: standard
+      run: vendor/bin/phpcs --config-set installed_paths ${{ github.workspace }}/standard/vendor/magento/magento-coding-standard,${{ github.workspace }}/standard/vendor/phpcompatibility/php-compatibility
+
+    - name: Get all changed files
+      id: changed-files
+      uses: tj-actions/changed-files@v35
+
+    - name: Checkout - Before Merge
+      shell: bash
+      working-directory: standard
+      id: previous-commit-files
+      run: |
+        if ${{ github.event_name == 'pull_request' }}; then
+          git checkout ${{ github.event.pull_request.base.ref }}
+        else
+          git checkout ${{ github.event.before }}
+        fi
+
+        for file in ${{ steps.changed-files.outputs.all_modified_files }}; do
```

Comment:
Why do we need to write this to a temp file? Couldn't we use `${{ steps.changed-files.outputs.all_modified_files }}` in the `phpcs` call directly?

Reply:
We are using --file-list options of PHPCS which required every file in the new line,  the all-modified-files having naming in the same line, so have to make those separate one.

Reply:
Okay. I think we could still use a shorter version without a temporary file. Some options like `sed` or `tr` are discussed [here](https://stackoverflow.com/q/1853009/719023).